bug: sudo aidevops approve fails on Linux — CLI wrapper and approval-helper use $HOME instead of SUDO_USER

## Description

`sudo aidevops approve issue <number>` fails completely on Linux with three cascading errors:

1. `[TIP] This project uses aidevops but isn't registered` — repos.json lookup uses `/root/.config/`
2. `[WARN] Agents not installed` — agents dir lookup uses `/root/.aidevops/`
3. `[ERROR] approval-helper.sh not found` — dispatch helper can't find the script

If the user works around (1-3) with `sudo HOME=/home/user ...`, a fourth error surfaces:

4. `[ERROR] gh authentication failed` — gnome-keyring is session-scoped via D-Bus, inaccessible to root

On macOS, none of these manifest because Keychain is system-wide and sudo doesn't aggressively reset HOME.

**Prior art:** #17754 (t1918) added `_require_gh_auth()` with a workaround message, but did not fix the root causes.

## Root Cause

Two independent bugs, both caused by sudo's `env_reset` on Linux:

**Bug 1 — Path resolution:** The CLI wrapper (lines 22-28) and `approval-helper.sh` (lines 26-29) use `$HOME` for all path variables (`AGENTS_DIR`, `INSTALL_DIR`, `CONFIG_DIR`, `APPROVAL_DIR`). Under sudo, `HOME=/root/`, so every path resolves to a non-existent location. The approval helper's `_approve_target()` already handles this correctly for the signing key via `SUDO_USER` + `_approval_real_home()` (line 130-134), but the CLI wrapper and the helper's global variables don't.

**Bug 2 — Keyring inaccessibility:** `gh auth status` uses gnome-keyring via D-Bus. Sudo strips `DBUS_SESSION_BUS_ADDRESS`, severing access to the keyring daemon. Even manually passing `GH_TOKEN` via `sudo VAR=val cmd` fails because `env_reset` strips it. The user's manual workaround `sudo GH_TOKEN=$(gh auth token) ...` also fails for the same reason.

## Fix (tested, working)

### CLI wrapper (`aidevops` CLI, lines 22-28)

Resolve real user's home via `getent passwd` when running under sudo:

```bash
# When running under sudo on Linux, HOME is reset to /root/ by env_reset.
# Resolve the real user's home via SUDO_USER so agent/config paths are correct.
# Security: no escalation — root already has full filesystem access.
if [[ -n "${SUDO_USER:-}" && "$(id -u)" -eq 0 ]]; then
    _AIDEVOPS_REAL_HOME=$(getent passwd "$SUDO_USER" | cut -d: -f6)
else
    _AIDEVOPS_REAL_HOME="$HOME"
fi
INSTALL_DIR="$_AIDEVOPS_REAL_HOME/Git/aidevops"
AGENTS_DIR="$_AIDEVOPS_REAL_HOME/.aidevops/agents"
CONFIG_DIR="$_AIDEVOPS_REAL_HOME/.config/aidevops"
REPOS_FILE="$CONFIG_DIR/repos.json"
VERSION_FILE="$INSTALL_DIR/VERSION"
```

### approval-helper.sh — global path variables (lines 26-29)

Same pattern for `APPROVAL_DIR`, `APPROVAL_PUB`, etc.:

```bash
if [[ -n "${SUDO_USER:-}" && "$(id -u)" -eq 0 ]]; then
    _APPROVAL_HOME=$(getent passwd "$SUDO_USER" | cut -d: -f6)
else
    _APPROVAL_HOME="$HOME"
fi
readonly APPROVAL_DIR="$_APPROVAL_HOME/.aidevops/approval-keys"
```

Also fix `_detect_slug()` repos.json path (line 44) to use `$_APPROVAL_HOME`.

### approval-helper.sh — `_require_gh_auth()` auto-resolve token

Replace the current error-only function with automatic resolution via `runuser` + D-Bus session reconnection:

```bash
_require_gh_auth() {
    if gh auth status >/dev/null 2>&1; then
        return 0
    fi
    if [[ -n "${SUDO_USER:-}" && "$(id -u)" -eq 0 ]]; then
        local real_uid
        real_uid=$(id -u "$SUDO_USER")
        # Method 1: runuser with reconstructed D-Bus session
        if [[ -S "/run/user/${real_uid}/bus" ]]; then
            local token
            token=$(runuser -u "$SUDO_USER" -- env \
                "DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/${real_uid}/bus" \
                "XDG_RUNTIME_DIR=/run/user/${real_uid}" \
                gh auth token 2>/dev/null || echo "")
            if [[ -n "$token" ]]; then
                export GH_TOKEN="$token"
                if gh auth status >/dev/null 2>&1; then
                    return 0
                fi
            fi
        fi
        # Method 2: file-based gh token (non-keyring storage)
        local real_home
        real_home=$(_approval_real_home)
        local gh_hosts="${real_home}/.config/gh/hosts.yml"
        if [[ -f "$gh_hosts" ]]; then
            local file_token
            file_token=$(awk '/oauth_token:/{print $2; exit}' "$gh_hosts" 2>/dev/null || echo "")
            if [[ -n "$file_token" ]]; then
                export GH_TOKEN="$file_token"
                if gh auth status >/dev/null 2>&1; then return 0; fi
            fi
        fi
    fi
    _print_error "gh authentication failed"
    _print_info "Workaround: export GH_TOKEN=\$(gh auth token) && sudo --preserve-env=GH_TOKEN aidevops approve ..."
    return 1
}
```

**Security analysis:** Root already has full filesystem and process access. Reading the user's GH token via `runuser` or their config file is not a privilege escalation. The token is held in memory only for the command duration. The D-Bus socket at `/run/user/<UID>/bus` is accessible to root by filesystem permissions. No secrets are persisted.

**Dependencies:** `getent` (glibc, universal), `runuser` (util-linux, universal on Linux). Both are part of base system on all major distributions.

## Steps to Reproduce

```bash
# On Linux (Ubuntu 24.04, any systemd distro with gnome-keyring)
sudo aidevops approve issue 12345 owner/repo
# Result: cascading errors, no approval posted
```

## Expected Behavior

`sudo aidevops approve issue <number>` should work on Linux identically to macOS — resolve paths via `SUDO_USER`, auto-resolve `gh` auth token, sign and post the approval.

## Environment

- **aidevops version**: 3.8.6
- **OS**: Ubuntu 24.04.4 LTS
- **Shell**: bash 5.2.21(1)-release
- **gh CLI**: 2.89.0
- **Auth storage**: gnome-keyring (keyring)
- **D-Bus socket**: `/run/user/1000/bus` (present, accessible)

## Additional Context

- Cosmetic bug in `cmd_status()`: `stat -f '%Su'` is macOS syntax; on Linux it dumps filesystem info instead of failing, so the `|| stat -c '%U'` fallback never triggers. The permissions check always shows "may be insecure" even when correct.
- Fix has been applied locally and tested successfully — all five issue approvals completed.

--- a/.agents/scripts/approval-helper.sh
+++ b/.agents/scripts/approval-helper.sh
@@ -23,7 +23,25 @@
 
 set -euo pipefail
 
-readonly APPROVAL_DIR="$HOME/.aidevops/approval-keys"
+# Resolve the real user's home directory, handling sudo env_reset on Linux.
+# Under sudo on Linux, HOME is reset to /root/ by env_reset; SUDO_USER holds
+# the original username. getent passwd is the canonical resolver on Linux.
+# On macOS, sudo does not reset HOME (and getent is not available), so the
+# fallback to $HOME is correct for both platforms.
+# Security: no escalation — root already has full filesystem access.
+_resolve_real_home() {
+	if [[ -n "${SUDO_USER:-}" && "$(id -u)" -eq 0 ]] && command -v getent &>/dev/null; then
+		getent passwd "$SUDO_USER" | cut -d: -f6
+	else
+		printf '%s' "$HOME"
+	fi
+	return 0
+}
+
+# Compute real home once at script load; used for all path variables below.
+_APPROVAL_HOME=$(_resolve_real_home)
+
+readonly APPROVAL_DIR="$_APPROVAL_HOME/.aidevops/approval-keys"
 readonly APPROVAL_PRIVATE_DIR="$APPROVAL_DIR/private"
 readonly APPROVAL_KEY="$APPROVAL_PRIVATE_DIR/approval.key"
 readonly APPROVAL_PUB="$APPROVAL_DIR/approval.pub"
@@ -41,7 +59,7 @@ _detect_slug() {
 	fi
 	# Fall back to repos.json current directory match
 	if [[ -z "$slug" || "$slug" != *"/"* ]]; then
-		local repos_json="$HOME/.config/aidevops/repos.json"
+		local repos_json="$_APPROVAL_HOME/.config/aidevops/repos.json"
 		if [[ -f "$repos_json" ]]; then
 			local cwd
 			cwd=$(pwd)
@@ -79,12 +97,47 @@ _print_error() {
 }
 
 _require_gh_auth() {
-	if ! gh auth status >/dev/null 2>&1; then
-		_print_error "gh authentication failed (common under sudo on Linux — gnome-keyring is inaccessible)"
-		_print_info "Workaround: export GH_TOKEN before sudo, or run: sudo GH_TOKEN=\$(gh auth token) aidevops approve ..."
-		return 1
+	if gh auth status >/dev/null 2>&1; then
+		return 0
 	fi
-	return 0
+	# Under sudo on Linux, gnome-keyring is inaccessible because env_reset strips
+	# DBUS_SESSION_BUS_ADDRESS. Attempt automatic token resolution via two methods
+	# before falling back to a descriptive error.
+	if [[ -n "${SUDO_USER:-}" && "$(id -u)" -eq 0 ]]; then
+		local real_uid=""
+		real_uid=$(id -u "$SUDO_USER" 2>/dev/null || echo "")
+		# Method 1: reconnect to user's D-Bus session socket via runuser
+		if [[ -n "$real_uid" && -S "/run/user/${real_uid}/bus" ]] && command -v runuser &>/dev/null; then
+			local token=""
+			token=$(runuser -u "$SUDO_USER" -- env \
+				"DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/${real_uid}/bus" \
+				"XDG_RUNTIME_DIR=/run/user/${real_uid}" \
+				gh auth token 2>/dev/null || echo "")
+			if [[ -n "$token" ]]; then
+				export GH_TOKEN="$token"
+				if gh auth status >/dev/null 2>&1; then
+					return 0
+				fi
+			fi
+		fi
+		# Method 2: read token directly from gh config file (non-keyring storage)
+		local real_home
+		real_home=$(_resolve_real_home)
+		local gh_hosts="${real_home}/.config/gh/hosts.yml"
+		if [[ -f "$gh_hosts" ]]; then
+			local file_token=""
+			file_token=$(awk '/oauth_token:/{print $2; exit}' "$gh_hosts" 2>/dev/null || echo "")
+			if [[ -n "$file_token" ]]; then
+				export GH_TOKEN="$file_token"
+				if gh auth status >/dev/null 2>&1; then
+					return 0
+				fi
+			fi
+		fi
+	fi
+	_print_error "gh authentication failed (common under sudo on Linux — gnome-keyring is inaccessible)"
+	_print_info "Workaround: export GH_TOKEN=\$(gh auth token) && sudo --preserve-env=GH_TOKEN aidevops approve ..."
+	return 1
 }
 
 _require_number_arg() {
@@ -128,9 +181,7 @@ _approval_real_user() {
 }
 
 _approval_real_home() {
-	local real_user
-	real_user=$(_approval_real_user)
-	eval echo "~$real_user"
+	_resolve_real_home
 	return 0
 }
 
@@ -464,7 +515,7 @@ cmd_setup() {
 	# Detect the real user behind sudo
 	local real_user="${SUDO_USER:-$(whoami)}"
 	local real_home
-	real_home=$(eval echo "~$real_user")
+	real_home=$(_resolve_real_home)
 	local actual_approval_dir="$real_home/.aidevops/approval-keys"
 	local actual_private_dir="$actual_approval_dir/private"
 	local actual_key="$actual_private_dir/approval.key"

--- a/aidevops.sh
+++ b/aidevops.sh
@@ -19,9 +19,17 @@ BOLD='\033[1m'
 NC='\033[0m' # No Color
 
 # Paths
-INSTALL_DIR="$HOME/Git/aidevops"
-AGENTS_DIR="$HOME/.aidevops/agents"
-CONFIG_DIR="$HOME/.config/aidevops"
+# When running under sudo on Linux, HOME is reset to /root/ by env_reset.
+# Resolve the real user's home via SUDO_USER so agent/config paths are correct.
+# Security: no escalation — root already has full filesystem access.
+if [[ -n "${SUDO_USER:-}" && "$(id -u)" -eq 0 ]]; then
+	_AIDEVOPS_REAL_HOME=$(getent passwd "$SUDO_USER" | cut -d: -f6)
+else
+	_AIDEVOPS_REAL_HOME="$HOME"
+fi
+INSTALL_DIR="$_AIDEVOPS_REAL_HOME/Git/aidevops"
+AGENTS_DIR="$_AIDEVOPS_REAL_HOME/.aidevops/agents"
+CONFIG_DIR="$_AIDEVOPS_REAL_HOME/.config/aidevops"
 REPOS_FILE="$CONFIG_DIR/repos.json"
 # shellcheck disable=SC2034  # Used in fresh install fallback
 REPO_URL="https://github.com/marcusquinn/aidevops.git"

